bug: release-please auto-dispatch fails — 'not a git repository'

## Summary

The auto-dispatch fix from #65 / PR #66 does not work. The \`Dispatch PyPI publish workflow\` step in \`.github/workflows/release-please.yml\` fails because the job does not check out the repo, and \`gh workflow run\` requires git context.

## Evidence

Run [24305105984](https://github.com/wgergely/vaultspec-core/actions/runs/24305105984) (merge of release-please PR #68, which created v0.1.10):

\`\`\`
Dispatching publish.yml for tag: vaultspec-core-v0.1.10
failed to run git: fatal: not a git repository (or any of the parent directories): .git
##[error]Process completed with exit code 1.
\`\`\`

v0.1.10 was released but publish never fired. PyPI stayed at v0.1.9 until manual dispatch.

## Fix

Add an \`actions/checkout\` step before the dispatch step, **or** use the REST API directly via \`gh api\` instead of \`gh workflow run\` (REST API doesn't need git context).

Preferred: use \`gh api\` — avoids the unnecessary checkout.

\`\`\`yaml
- name: Dispatch PyPI publish workflow
  if: steps.release.outputs.release_created == 'true'
  env:
    GH_TOKEN: \${{ secrets.GITHUB_TOKEN }}
    TAG: \${{ steps.release.outputs.tag_name }}
  run: |
    gh api --method POST \
      "repos/\${GITHUB_REPOSITORY}/actions/workflows/publish.yml/dispatches" \
      -f "ref=\${TAG}" \
      -f "inputs[tag]=\${TAG}"
\`\`\`

## Immediate action

Manually dispatch publish for v0.1.10 (done separately).

## Related

- #65 (original PyPI auto-publish bug)
- PR #66 (the fix that broke in a different way)

--- a/.github/workflows/release-please.yml
+++ b/.github/workflows/release-please.yml
@@ -61,17 +61,8 @@ jobs:
             exit 1
           fi
           echo "Dispatching publish.yml for tag: ${TAG}"
-          gh workflow run publish.yml --ref "${TAG}" -f tag="${TAG}"
-          echo "Waiting for dispatched run to appear..."
-          for _ in 1 2 3 4 5 6 7 8 9 10; do
-            RUN_ID=$(gh run list --workflow=publish.yml --event=workflow_dispatch --limit=1 --json databaseId,headBranch \
-              --jq ".[] | select(.headBranch==\"${TAG}\") | .databaseId")
-            if [ -n "${RUN_ID}" ]; then
-              echo "Dispatched run: ${RUN_ID}"
-              echo "https://github.com/${GITHUB_REPOSITORY}/actions/runs/${RUN_ID}" >> "$GITHUB_STEP_SUMMARY"
-              exit 0
-            fi
-            sleep 3
-          done
-          echo "::error::publish.yml dispatch did not produce a visible run within 30s"
-          exit 1
+          gh api --method POST \
+            "repos/${GITHUB_REPOSITORY}/actions/workflows/publish.yml/dispatches" \
+            -f "ref=${TAG}" \
+            -f "inputs[tag]=${TAG}"
+          echo "Dispatched publish.yml for ${TAG}" >> "$GITHUB_STEP_SUMMARY"

